src/main/java/com/fruits/ecommerce/Configuration/SecurityConfig/ExtraServices/LoginAttemptService.java

```diff
@@ -0,0 +1,63 @@
+package com.fruits.ecommerce.Configuration.SecurityConfig.ExtraServices;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import jakarta.annotation.Nonnull;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+
+@Slf4j
+@Service
+public class LoginAttemptService {
```

Comment:
I try to use best practices for my code and I use a service like LoginAttemptService that has several features: 1. Protection from brute force attacks and 2. Account Locking: If the service notices that there are a large number of failed login attempts, it can temporarily lock the account for a specified period, which increases the security of the system and prevents attackers from continuing to try.  3. Sending alert notifications    4. Improving the security attempt log
 5. Integration with Spring Security